napari/movie/naparimovie.py

```diff
@@ -0,0 +1,364 @@
+import copy
+
+from . import util
+
+
+class Movie:
+    def __init__(self, myviewer=None, inter_steps=15):
+
+        """Standard __init__ method.
+
+        Parameters
+        ----------
+        myviewer : napari viewer
+            napari viewer
+        inter_steps: int
+            number of steps to interpolate between key frames
+
+        Attributes
+        ----------
+        key_frames : list
+            list of dictionary defining napari viewer states. Dictionaries have keys:
+                'frame': int, frame
+                'ndisplay': int, view type (2,3)
+                'camera': camera state
+                'vis': list of booleans, visibility of layers
+                'sliders': list, slider positions
+
+        interpolated_states: dict
+            dictionary defining interpolated states. Each element is a list of length N
+            frames. Keys are:
+                'ndisplay': int, view type (2,3)
+                'camera': camera state
+                'vis': list of booleans, visibility of layers
+                'sliders': list, slider positions
+
+        states_dict : list
+            list of dictionaries defining napari viewer states for each frame. Same keys as key_frames
+
+        current_frame : int
+            currently shown key frame
+        implot : matplotlib Ax object
+            reference to matplotlib image used for movie returned by imshow
+        anim : matplotlib FuncAnimation object
+            reference to animation object
+        """
+
+        if myviewer is None:
+            raise TypeError(
+                "You need to pass a napari viewer for the myviewer argument"
+            )
+        else:
+            self.myviewer = myviewer
+
+        self.key_frames = []
+        self.inter_steps = inter_steps
+
+        self.current_frame = -1
+        self.current_interpolframe = 0
+
+        # establish key bindings
+        self.add_callback()
+
+    def finish_movie(self):
+        self.release_callbacks()
+
+    def add_callback(self):
+        """Bind keys"""
+
+        self.myviewer.bind_key("f", self.capture_keyframe_callback)
+        self.myviewer.bind_key("r", self.replace_keyframe_callback)
+        self.myviewer.bind_key("d", self.delete_keyframe_callback)
+
+        self.myviewer.bind_key("a", self.key_adv_frame)
+        self.myviewer.bind_key("b", self.key_back_frame)
+
+        self.myviewer.bind_key("w", self.key_interpolframe)
+
+    def release_callbacks(self):
+        """Release keys"""
+
+        self.myviewer.bind_key("f", None)
+        self.myviewer.bind_key("r", None)
+        self.myviewer.bind_key("d", None)
+
+        self.myviewer.bind_key("a", None)
+        self.myviewer.bind_key("b", None)
+
+        self.myviewer.bind_key("w", None)
+
+    def get_new_state(self):
+        """Capture current napari state
+
+        Returns
+        -------
+        new_state : dict
+            description of state
+        """
+
+        new_state = {
```

Comment:
As noted in the discussion in https://github.com/napari/napari/issues/448#issuecomment-563994868 once #686 goes in we can start using the serialization of the viewer state, but for now this is fine for me, though mayb @jni has preferrences